File: 1678.设计-goal-解析器.go
/*
 * @lc app=leetcode.cn id=1678 lang=golang
 *
 * [1678] 设计 Goal 解析器
 */

// @lc code=start
func interpret(command string) string {
	// return strings.Replace(strings.Replace(command, "()", "o", -1), "(al)", "al", -1)
	preLetter := command[0]
	index := 1
	res := make([]byte, 0, len(command))
	if preLetter == 'G' {
		res = append(res, 'G')
	}
	for index < len(command) {
		if command[index] == ')' && preLetter == '(' {
			res = append(res, 'o')
		} else if command[index] == 'a' {
			preLetter = ')'
			index += 3
			res = append(res, []byte("al")...)
			continue
		} else if command[index] == 'G' {
			res = append(res, 'G')
		}
		preLetter = command[index]
		index++
	}
	return string(res[:len(res)])
}

// @lc code=end

File: 1678.设计-goal-解析器.py
#
# @lc app=leetcode.cn id=1678 lang=python3
#
# [1678] 设计 Goal 解析器
#

# @lc code=start
class Solution:
    def interpret(self, command: str) -> str:
        # return command.replace("()", "o").replace("(al)", "al")
        pre_letter = command[0]
        res = []
        index = 1
        if pre_letter == "G":
            res.append("G")
        while index < len(command):
            if command[index] == ")" and pre_letter == "(":
                res.append("o")
            elif command[index] == "a":
                pre_letter = command[index]
                index += 3
                res.append("al")
                continue
            elif command[index] == "G":
                res.append("G")
            pre_letter = command[index]
            index += 1
        return "".join(res)
# @lc code=end
